src/iap/shared/validators/iap-sync-validators.validator.ts

```diff
@@ -0,0 +1,22 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+export class IapSyncValidators {
+	static matchPassword(password: string, passwordConfirmation: string): ValidatorFn {
+		return (control: AbstractControl): ValidationErrors | null => {
+			const passwordControl = control.get(password);
+			const passwordConfirmationControl = control.get(passwordConfirmation);
+
+			if (!passwordConfirmationControl || !passwordControl) {
+				return null;
+			}
+
+			if (passwordControl.value !== passwordConfirmationControl.value) {
+				passwordConfirmationControl.setErrors({ passwordMismatch: true });
```

Comment:
Developers usually return the validator name as a property of ValidationErrors object. It helps to understand more quickly what sort of error happened.
```suggestion
				passwordConfirmationControl.setErrors({ matchPassword: true });
```